### PR TITLE
Packet of inventory expansion feature.

### PR DESCRIPTION
**Pull request by sctnightcore**
issues #2587

### CODE REVIEW

File: src/Network/Receive/kRO/Sakexe_0.pm [hunk 1]
```diff
@@ -3638,5 +3640,34 @@ sub progress_bar_unit {
...
+}
+
+#expand_inventory_result 
+use constant {
+	EXPAND_INVENTORY_RESULT_SUCCESS = 0x0,
+	EXPAND_INVENTORY_RESULT_FAILED = 0x1,
+	EXPAND_INVENTORY_RESULT_OTHER_WORK = 0x2,
+	EXPAND_INVENTORY_RESULT_MISSING_ITEM = 0x3,
+	EXPAND_INVENTORY_RESULT_MAX_SIZE = 0x4	
+}
```
Comment (sctnightcore) [inline src/Network/Receive/kRO/Sakexe_0.pm]: miss ;

File: src/Network/Receive/kRO/Sakexe_0.pm [hunk 2]
```diff
@@ -3638,5 +3640,34 @@ sub progress_bar_unit {
 	debug "Displays progress bar (GID: $args-{GID} time: $args-{time})\n";	
 }
 
+sub inventory_expansion_info {
+	my($self, $args) = @_;
+	#sd->inventorySize - FIXED_INVENTORY_SIZE; 
+	#hardcode inventorysize ? [sctnightcore]
```
Comment (sctnightcore) [inline src/Network/Receive/kRO/Sakexe_0.pm]: My idea : 
my $size = $char->{'weight_max'} - $args->{expansionSize};
message TF("You have more storage space %s",$size),"info";

File: src/Network/Send/kRO/Sakexe_0.pm [hunk 3]
```diff
@@ -757,4 +759,13 @@ sub sendPartyOrganize {
 # 0x0206,11
 # 0x0207,34
 
+sub Send_Inventory_Expansion_Request {
```
Comment (alisonrag) [inline src/Network/Send/kRO/Sakexe_0.pm]: please keep the pattern

rename this
Send_Inventory_Expansion_Request

to this:
sendInventoryExpansionRequest

File: src/Network/Send/kRO/Sakexe_0.pm [hunk 4]
```diff
@@ -757,4 +759,13 @@ sub sendPartyOrganize {
...
+	my ($self, $args) = @_;
+	$self->sendToServer($self->reconstruct({ switch => 'inventory_expansion_request' }));
+}
+
+sub Send_Inventory_Expansion_Rejected {
```
Comment (alisonrag) [inline src/Network/Send/kRO/Sakexe_0.pm]: also here

Comment (sctnightcore) [reply src/Network/Send/kRO/Sakexe_0.pm]: fix : d3c42f0

File: src/Network/Send/kRO/Sakexe_0.pm (hunk 3 above)
Comment (sctnightcore) [reply src/Network/Send/kRO/Sakexe_0.pm]: fix : d3c42f0

File: src/Network/Receive/kRO/Sakexe_0.pm [hunk 5]
```diff
@@ -2265,6 +2267,41 @@ sub party_dead {
 	}	
 }
 
+sub progress_bar_unit {
+	my($self, $args) = @_;
+	debug "Displays progress bar (GID: $args-{GID} time: $args-{time})\n";
```
Comment (alisonrag) [inline src/Network/Receive/kRO/Sakexe_0.pm]: ` $args-{GID} and $args-{time}`

fix it to

` $args->{GID} and $args->{time}`

File: src/Network/Receive/kRO/Sakexe_0.pm [hunk 6]
```diff
@@ -637,6 +637,8 @@ sub new {
 		'0AE4' => ['party_join', 'a4 a4 V v4 C Z24 Z24 Z16 C2', [qw(ID charID role jobID lv x y type name user map item_pickup item_share)]],
 		'0AE5' => ['party_users_info', 'v Z24 a*', [qw(len party_name playerInfo)]],
 		'0AFD' => ['sage_autospell', 'v a*', [qw(len autospell_list)]], #herc PR 2310
+		'0B18' => ['inventory_expansion_info', 'v', [qw(expansionSize)]], # expansionSize = inventorySize [sctnightcore]
+		'0B18' => ['inventory_expansion_result', 'v', [qw(result)]], #
```
Comment (ya4ept) [inline src/Network/Receive/kRO/Sakexe_0.pm:641]: wtf? may be
```
'0B17' => ['inventory_expansion_result', 'v', [qw(result)]], #
```

**Pull request merged**